assets/js/modules/adsense/util/status.js

```diff
@@ -0,0 +1,251 @@
+ * @param {?Array}  data.alerts            List of alert objects retrieved from the API.
+ * @param {?Object} data.error             Error object if one of the API requests failed.
+ * @param {?string} data.previousAccountID Account ID, if already known from before.
+ * @param {?string} data.previousClientID  Client ID, if already known from before.
+ * @return {?string} Account status determined, or undefined if one of the required
+ *                   parameters is undefined.
+ */
+export const determineAccountStatus = ( {
+	accounts,
+	clients,
+	alerts,
+	error,
+	previousAccountID,
+	previousClientID,
+} ) => {
+	if ( 'undefined' === typeof accounts || 'undefined' === typeof previousAccountID ) {
+		return errorToStatus( error );
+	}
+
+	const accountID = determineAccountID( { accounts, previousAccountID } );
+	if ( ! accountID ) {
+		// If there are accounts, but the account ID cannot be determined, it
+		// means that there are multiple accounts and the user needs to select
+		// one.
+		if ( accounts.length ) {
+			return ACCOUNT_STATUS_MULTIPLE;
+		}
+		return ACCOUNT_STATUS_NONE;
+	}
+
+	if ( 'undefined' === typeof alerts ) {
+		return errorToStatus( error );
+	}
+
+	const hasGraylistedAlert = alerts.some( ( alert ) => {
+		return 'GRAYLISTED_PUBLISHER' === alert.type;
+	} );
+	if ( hasGraylistedAlert ) {
+		return ACCOUNT_STATUS_GRAYLISTED;
+	}
+
+	if ( 'undefined' === typeof clients || 'undefined' === typeof previousClientID ) {
+		return errorToStatus( error );
+	}
+
+	const clientID = determineClientID( { clients, previousClientID } );
+	if ( ! clientID ) {
+		return ACCOUNT_STATUS_NO_CLIENT;
+	}
+
+	return ACCOUNT_STATUS_APPROVED;
+};
+
+/**
+ * Determines the AdSense site status for given input data.
+ *
+ * This utility function should be used in combination with data retrieved from
+ * the datastore, hence passing undefined (loading state) is supported.
+ *
+ * The AdSense API currently only surfaces limited information about the approval
+ * status for a site, so at this point it is only possible to know whether the
+ * site has been added or not. In other words, the site being added does not
+ * necessarily mean it has been already approved - there still may be actions
+ * required for the user.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}  data             Input data to determine site status.
+ * @param {?Array}  data.urlChannels List of URL channel objects retrieved from the API.
+ * @param {?string} data.siteURL     URL of this website.
+ * @return {?string} Site status determined, or undefined if one of the required
+ *                   parameters is undefined.
+ */
+export const determineSiteStatus = ( {
+	urlChannels,
+	siteURL,
+} ) => {
+	if ( 'undefined' === typeof urlChannels || 'undefined' === typeof siteURL ) {
+		return undefined;
+	}
+
+	const hasSiteURL = urlChannels.some( ( urlChannel ) => {
+		return 0 <= siteURL.indexOf( urlChannel.urlPattern );
```

Comment:
This is minor, but idiomatically I usually see this written as:
```suggestion
		return siteURL.indexOf( urlChannel.urlPattern ) !== -1;
```
in JavaScript.

I guess partly it's the WordPress/"Yoda"-style here and partially the `<= 0` instead of `-1 !==` that's throwing me off, but collectively this made me do a double-take. 🤷‍♂

Reply:
I'll just address this in the next one :)